addons/mail/static/src/discuss/core/common/discuss_core_common_service.js

```diff
@@ -90,24 +90,27 @@ export class DiscussCoreCommon {
             if (message.isSelfAuthored) {
                 thread.onNewSelfMessage(message);
             } else {
-                if (thread.isDisplayed && thread.self_member_id?.new_message_separator_ui === 0) {
-                    thread.self_member_id.new_message_separator_ui = message.id;
+                if (
+                    thread.isDisplayed &&
+                    thread.channel?.self_member_id?.new_message_separator_ui === 0
+                ) {
+                    thread.channel.self_member_id.new_message_separator_ui = message.id;
                 }
-                if (!thread.isDisplayed && thread.self_member_id) {
+                if (!thread.isDisplayed && thread.channel?.self_member_id) {
```

Comment:
This function should be changed to use channel as a local variable, not thread. We necessarily have a thread.